File: rust-typescript/projects/enums-2.ts
type Custom = {
  age: number;
  name: string;
};

type Item = number | string | Custom;

function append(items: Item[]) {
  items.push("Hello FEM");
}

const arrayItems: Item[] = [1, "two", { age: 123, name: "Jason" }];
console.log(arrayItems);
append(arrayItems);
console.log(arrayItems);

const arrayNumbers: number[] = [4, 5, 6];
console.log(arrayNumbers);
append(arrayNumbers);
console.log(arrayNumbers);

File: rust-typescript/projects/enums-2.rs
struct Custom {
    age: usize,
    name: String,
}

// Enums can have SubTypes
// Added Type discrimination in a way in which there
// is no way to have type discrimination
enum Item {
    Number(usize),
    String(String),
    MyCustom(Custom),
}

fn append(items: &mut Vec<Item>) {
    items.push(Item::String("Hello FEM".into()));
}

fn main() {
    let mut items: Vec<Item> = vec![];
    append(&mut items);
}
